plugins/modules/device_credential_workflow_manager.py

```diff
@@ -2087,214 +2100,232 @@ def get_want_assign_credentials(self, AssignCredentials):
                         return self
-                elif snmpV2cWriteDescription:
-                    for item in snmpV2cWrite_details:
-                        if item.get("description") == snmpV2cWriteDescription:
-                            snmpV2cWriteDetail = item
-                    if not snmpV2cWriteDetail:
+                elif snmp_v2c_write_description:
+                    for item in global_snmp_v2c_write_details:
+                        if item.get("description") == snmp_v2c_write_description:
+                            snmp_v2c_write_detail = item
+
+                    if not snmp_v2c_write_detail:
                         self.msg = "The username and description of the snmp_v2c_write credential are invalid."
                         self.status = "failed"
                         return self
+
                 if current_ccc_version_as_int <= self.get_ccc_version_as_int_from_str("2.3.5.3"):
-                    want.get("assign_credentials").update({"snmpV2WriteId": snmpV2cWriteDetail.get("id")})
+                    want.get("assign_credentials").update({"snmpV2WriteId": snmp_v2c_write_detail.get("id")})
                 else:
                     want.get("assign_credentials").update({
-                        "snmpv2cWriteCredentialsId": {"credentialsId": snmpV2cWriteDetail.get("id")}
+                        "snmpv2cWriteCredentialsId": {"credentialsId": snmp_v2c_write_detail.get("id")}
                     })
 
-        https_read = AssignCredentials.get("https_read")
+        https_read = assign_credentials.get("https_read")
         if https_read:
-            httpReadId = https_read.get("id")
-            httpReadDescription = https_read.get("description")
-            httpReadUsername = https_read.get("username")
-            if httpReadId or httpReadDescription and httpReadUsername:
+            http_read_id = https_read.get("id")
+            http_read_description = https_read.get("description")
+            http_read_username = https_read.get("username")
 
+            if http_read_id or http_read_description and http_read_username:
                 # All httpRead details from the Cisco Catalyst Center
-                httpRead_details = global_credentials.get("httpsRead")
-                if not httpRead_details:
-                    self.msg = "Global httpRead Credential is not available."
+                http_read_details = global_credentials.get("httpsRead")
+                if not http_read_details:
+                    self.msg = "Global http_read Credential is not available."
                     self.status = "failed"
                     return self
-                httpReadDetail = None
-                if httpReadId:
-                    httpReadDetail = get_dict_result(httpRead_details, "id", httpReadId)
-                    if not httpReadDetail:
-                        self.msg = "The ID of the httpRead credential is not valid."
+                http_read_detail = None
```

Comment:
Please note that this variable name should be different.
Earlier the code is httpRead_details  and  httpReadDetail. But now you changed both into
http_read_detail which is wrong...

Reply:
http_read_details = global_credentials.get("httpsRead")
For the above, we can use global_http_read_details ..

Reply:
No variable names were different only, it's http_read_detail and http_read_detail(s).
now http_read_details changed to global_https_read_details